app/controllers/users_controller.rb

```diff
@@ -46,16 +46,14 @@ def destroy
   end
 
   def filters
-    if(params[:search].present?)
-      if params[:search].eql? User::ROLES[0]
-        @users = @users.where(role: User::ROLES[0])
-      elsif params[:search].eql? User::ROLES[1]
-        @users = @users.where(role: User::ROLES[1])
-      elsif params[:search].eql? User::ROLES[2]
-        @users = @users.where(role: User::ROLES[2])
-      end
-    end
-    @pagy, @users = pagy(@users.order(created_at: :desc), items: Company::PAGE_SIZE)
+    search_options = {
+      id: @users.pluck(:id),
+      page: params[:page]
+    }
+    search_options[:role] = params[:search] if User::ROLES.include? params[:search]
+
+    @pagy, @users = User.filter_search(nil, search_options)
```

Comment:
The search method used within also provides searching like @users.search but that was giving a depreciation warning so instead i passed ids as a condition of where clause for the search